src/components/ProjectDashboard/components/NftRewardsPanel/hooks/nftRewards.ts

```diff
@@ -0,0 +1,100 @@
+import { PayProjectFormContext } from 'components/Project/PayProjectForm/payProjectFormContext'
+import { DEFAULT_ALLOW_OVERSPENDING } from 'constants/transactionDefaults'
+import { NftRewardsContext } from 'contexts/NftRewards/NftRewardsContext'
+import { CurrencyContext } from 'contexts/shared/CurrencyContext'
+import { useCurrencyConverter } from 'hooks/useCurrencyConverter'
+import { useContext } from 'react'
+import { fromWad } from 'utils/format/formatNumber'
+import { sumTierFloors } from 'utils/nftRewards'
+
+export const useNftRewards = () => {
+  const {
+    nftRewards: { rewardTiers },
+    loading,
+  } = useContext(NftRewardsContext)
+  const { form: payProjectForm } = useContext(PayProjectFormContext)
+  const {
+    currencies: { ETH },
+  } = useContext(CurrencyContext)
+
+  const converter = useCurrencyConverter()
+
+  const {
+    payAmount,
+    payMetadata,
+    payInCurrency,
+    setPayAmount,
+    setPayInCurrency,
+    setPayMetadata,
+    validatePayAmount,
+  } = payProjectForm ?? {}
+
+  const payAmountETH =
+    payInCurrency === ETH ? payAmount : fromWad(converter.usdToWei(payAmount))
+
+  const handleTierSelect = (
```

Comment:
consider wrapping this in a `useCallback`

Reply:
Good call, done